### DISCUSSION TITLE
Hi, so I mad a piece of code that when can output a incorrect answer.

### DISCUSSION THREAD

**Discussion by pigallen** [community/community#130821] [Programming Help]
### Body

Here is the code.

score = input("Enter a score between 0.0 and 1.0: ")

# Convert input to a float
try:
    score = float(score)
except ValueError:
    print("Error: Please enter a numeric value.")
    exit()
problem starts here
try:
    score > 1
except ValueError:
    print("RTFM!")
    exit()
ends here
# Check if score is within range and assign grade
if score >= 0.9 and score <= 1.0:
    grade = 'A'
elif score >= 0.8:
    grade = 'B'
elif score >= 0.7:
    grade = 'C'
elif score >= 0.6:
    grade = 'D'
elif score >= 0.0 and score < 0.6:
    grade = 'F'
else:
    print("Error: Score is out of range. Please enter a score between 0.0 and 1.0.")
    exit()

# Print the grade
print(f"{grade}")

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by tanyatiwari995** [comment]
Possible Issues in the Code
Redundant try-except block:

The second try-except block is unnecessary for checking if score > 1. This is a logical condition, not an operation that raises a ValueError.
Improper use of exit():

The exit() function is called inside the exception handlers, which may not be the best approach. Instead, the program could return an error message and prompt the user for a valid input.
Improper Exception Handling:

The second try-except block is used incorrectly. The condition score > 1 does not raise a ValueError, so the exception handler except ValueError: will never be executed.

score = input("Enter a score between 0.0 and 1.0: ")

# Convert input to float and check validity
try:
    score = float(score)
    if score < 0 or score > 1:
        print("Error: Score must be between 0.0 and 1.0.")
        exit()
except ValueError:
    print("Error: Please enter a numeric value.")
    exit()

# Assign grade based on score
if score >= 0.9:
    grade = "A"
elif score >= 0.8:
    grade = "B"
elif score >= 0.7:
    grade = "C"
elif score >= 0.6:
    grade = "D"
else:
    grade = "F"

print("Your grade is:", grade)
